### PR TITLE
Update Makefile and README.md

### PR DESCRIPTION
**Pull request by marksiemers**
Some make commands were failing for various reasons.
Fix Makefile to get commands working.

Update README to provide working examples using 'make'

### CODE REVIEW

File: Makefile
Comment (marksiemers) [inline Makefile:53]: Is it better to build the generator binary, then call it?

Crystal really provides both options - building first, or executing inline.

Comment (ErikSchierboom) [reply Makefile:53]: What is the downside of executing inline? As generators are only rarely being called, performance should probably not be an argument (not sure if it is).

Comment (hiljusti) [reply Makefile:53]: I'd either:

1. Run it in an interpreted mode
2. Give it a make rule (so you at least get "dirty bit" rebuilds)

But I think running interpreted is fine.

File: Makefile
Comment (hiljusti) [inline Makefile:47]:
```suggestion
.PHONY: build-generator
build-generator:
```

On the off-chance that someone might actually put a `build-generator` file or executable here, or rename the target to something that matches a file, this would prevent Make thinking it should only run this target if the file modification time was updated.

File: Makefile
Comment (hiljusti) [inline Makefile:51]:
```suggestion
.PHONY: generate-exercise
generate-exercise:
```

(Same reasoning as `build-generator`)

File: Makefile
Comment (marksiemers) [reply Makefile:53]: @hiljusti - Do you think it is too early to run in interpreted mode?

> The interpreter is currently on an experimental phase, with lots of [missing bits](https://github.com/crystal-lang/crystal/issues/11555). The reason to merge it at this early stage is to enable a proper discussion of interpreter-related PRs and speed up its development a bit. For those willing to try it out, we welcome interpreter-related issues taking into consideration the already known issues aforelinked.

Source: https://crystal-lang.org/2021/12/29/crystal-i.html

Comment (hiljusti) [reply Makefile:55]: Maybe, but idk, I'd wait until seeing a problem before changing it. It seems to be working fine so far, and it is just running a single file so if it's working now it might just keep working forever? (i.e. it's not some crazy huge thing with complex dependencies that would stress the edges of their interpreter)

**Review by hiljusti — APPROVED** [review]

**Comment by marksiemers** [comment]
@ErikSchierboom - I think this is ready to merge.

@hiljusti - I think some clarification is needed about running crystal, I would say there are three ways to run it:
1. "Inline" (unsure the real term) - `crystal your_file.cr`
2. "Interpreted - `crystal i your_file.cr` (I think this allows a REPL like experience)
3. "Compiled" - `crystal build --release your_file.cr && ./your_file`

I'm assuming and hoping we're all on board with using "inline" mode - since it is working and not a high risk of causing problems.

**Comment by ErikSchierboom** [comment]
@marksiemers I'll wait for @hiljusti's approval, as I don't really know Crystal

**Comment by hiljusti** [comment]
Let's do it!

Thanks for your patience, getting to these when I can.

**Pull request merged**
